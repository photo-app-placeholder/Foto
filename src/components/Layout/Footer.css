footer {
  text-align: center;

  bottom: 0;
  margin-top: 70px;

  width: 100%;
  height: 20px;
  color: rgb(153, 153, 153);
  font-size: 8pt;
}
