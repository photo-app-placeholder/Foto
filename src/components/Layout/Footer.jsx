import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Footer.css';

export default function Footer() {
  return <footer>Â© All Rights Reserved. 2022</footer>;
}
