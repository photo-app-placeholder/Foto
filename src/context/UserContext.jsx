import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

export const UserContext = createContext();

export const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  console.log(user);

  const contextValue = { user, setUser };

  return (
    <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>
  );
};

export const useUser = () => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('Error: useUser needs to be used within UserContext');
  }
  return context;
};
