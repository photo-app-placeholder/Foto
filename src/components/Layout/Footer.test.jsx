import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import Footer from './Footer';

test('should display footer', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
