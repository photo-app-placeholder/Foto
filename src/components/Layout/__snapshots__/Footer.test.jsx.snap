// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display footer 1`] = `
<div>
  <footer>
    Â© All Rights Reserved. 2022
  </footer>
</div>
`;
